src/components/sections/Newsletter/Newsletter.tsx

```diff
@@ -0,0 +1,82 @@
+import type { ComponentPropsWithRef, FormEvent, ReactNode } from 'react'
+import React, { forwardRef, useRef } from 'react'
+import { Form, Label, Input, Button } from '@faststore/ui'
+
+export interface NewsletterProps
+  extends Omit<ComponentPropsWithRef<'form'>, 'onSubmit'> {
+  /**
+   * Heading for the section.
+   */
+  heading: ReactNode
+  /**
+   * Description about the newsletter.
+   */
+  description?: ReactNode
+  /**
+   * Callback function when submitted.
+   */
+  onSubmit: (value: string) => void
+}
+
+const Newsletter = forwardRef<HTMLFormElement, NewsletterProps>(
+  function Newsletter({ heading, description, onSubmit, ...otherProps }, ref) {
+    const emailInputRef = useRef<HTMLInputElement>(null)
+
+    const handleSubmit = (event: FormEvent) => {
+      event.preventDefault()
+
+      if (emailInputRef.current?.value !== '') {
+        onSubmit(emailInputRef.current!.value)
+      }
+    }
+
+    return (
+      <Form
+        data-store-newsletter
+        ref={ref}
+        onSubmit={handleSubmit}
+        {...otherProps}
+      >
+        <div data-newsletter-content>
+          {heading}
+          {Boolean(description) && description}
+        </div>
+
+        <div data-newsletter-controls>
```

Comment:
And this will be the `data-newsletter-content` ? What do you think?

Reply:
I kept it this as it is. See my explanation in [this previous comment](https://github.com/vtex-sites/base.store/pull/242/files#r795839671).